### PR TITLE
feat: add email OTP support

### PR DESCRIPTION
**Pull request by dphilipson**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR introduces support for OTP (One-Time Password) authentication in the existing email authentication flow. It updates various types, interfaces, and components to accommodate the new OTP mode, enhancing the user authentication experience.

### Detailed summary
- Added `otp` type to `Session` and `AuthParams`.
- Introduced `emailMode` option in authentication parameters.
- Updated UI components to handle OTP input and display.
- Modified error handling for OTP verification.
- Enhanced event listeners to support OTP connections.
- Updated documentation and examples for OTP flow.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: account-kit/signer/src/session/manager.ts [hunk 1]
```diff
@@ -7,7 +7,12 @@ import {
 } from "zustand/middleware";
 import { createStore, type Mutate, type StoreApi } from "zustand/vanilla";
 import type { BaseSignerClient } from "../client/base";
-import type { User } from "../client/types";
+import type {
+  AlchemySignerClientEvent,
+  AlchemySignerClientEvents,
+  AuthenticatingEventMetadata,
```
Comment (moldy530) [inline account-kit/signer/src/session/manager.ts:13]: ⚠️ **[eslint]** <[@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'AuthenticatingEventMetadata' is defined but never used.

File: account-kit/signer/src/session/manager.ts [hunk 2]
```diff
@@ -230,28 +244,45 @@ export class SessionManager {
...
     );
 
-    this.client.on("disconnected", () => this.clearSession());
-
-    this.client.on("connectedEmail", (user, bundle) =>
-      this.setSessionWithUserAndBundle({ type: "email", user, bundle })
-    );
+    // Helper type to ensure that a listener is either defined or explicitly
+    // omitted for every event type.
+    type Listeners = {
```
Comment (dphilipson) [inline account-kit/signer/src/session/manager.ts:249]: @moldy530 I kept forgetting to add listeners for new events I added so I changed this so the type system would help. Style question: do you prefer using `undefined` for events where we don't want a listener, or the noop function `() => {}`?

Comment (moldy530) [reply account-kit/signer/src/session/manager.ts:249]: oh good question... I think undefined is simpler devex, but only slightly. so I don't have a super strong feeling here

File: account-kit/signer/src/client/index.ts [hunk 3]
```diff
@@ -217,6 +218,45 @@ export class AlchemySignerWebClient extends BaseSignerClient<ExportWalletParams>
...
+   *   iframeContainerId: "signer-iframe-container",
+   *  },
+   * });
+   *
+   * const account = await client.submitOtpCode({
+   *   orgId: "user-org-id",
+   *   bundle: "bundle-from-email",
+   *   otpId: "opt-returned-from-initEmailAuth",
+   *   otpCode: "otp-code-from-email",
+   * });
```
[the remark this answers is not in the document]
Comment (dphilipson) [reply account-kit/signer/src/client/index.ts]: Good catch, fixed.

File: account-kit/signer/src/base.ts [hunk 4]
```diff
@@ -673,26 +683,30 @@ export abstract class BaseAlchemySigner<TClient extends BaseSignerClient>
...
             email: params.email,
+            emailMode: params.emailMode,
             expirationSeconds,
             redirectParams: params.redirectParams,
           });
 
-      this.sessionManager.setTemporarySession({ orgId });
+      this.sessionManager.setTemporarySession({
+        orgId,
+        isNewUser: !existingUser,
```
Comment (moldy530) [inline account-kit/signer/src/base.ts:705]: oh nice

**Review by moldy530 — APPROVED** [review]

**Pull request merged**